Source/VMware.vSphereDSC/Tests/Integration/VMHostService.Integration.Tests.ps1

```diff
@@ -0,0 +1,269 @@
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Name,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Server,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $User,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Password
+)
+
+$script:dscResourceName = 'VMHostService'
+$script:dscConfig = $null
+$script:moduleFolderPath = (Get-Module VMware.vSphereDSC -ListAvailable).ModuleBase
+$script:integrationTestsFolderPath = Join-Path (Join-Path $moduleFolderPath 'Tests') 'Integration'
+$script:configurationFile = "$script:integrationTestsFolderPath\Configurations\$($script:dscResourceName)\$($script:dscResourceName)_Config.ps1"
+
+$script:configWithServicePolicyUnset = "$($script:dscResourceName)_WithServicePolicyUnset_Config"
+$script:configWithServicePolicyOn = "$($script:dscResourceName)_WithServicePolicyOn_Config"
+$script:configWithServicePolicyOff = "$($script:dscResourceName)_WithServicePolicyOff_Config"
+$script:configWithServicePolicyOnAndRunning = "$($script:dscResourceName)_WithServicePolicyOnAndRunning_Config"
+
+$script:connection = Connect-VIServer -Server $Server -User $User -Password $Password
+$script:vmHost = $null
+
+$script:Key = 'TMS-SSH'
+$script:PolicyUnset = 'Unset'                   # Should be able to use [ServicePolicy]::Unset
+$script:PolicyOff = 'Off'                       # Should be able to use [ServicePolicy]::Off
+$script:PolicyOn = 'On'                         # Should be able to use [ServicePolicy]::On
+$script:RunningFalse = $false
+$script:RunningTrue = $true
+
+$script:resourceWithServicePolicyUnset = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyUnset
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOn = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOn
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOff = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOff
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOnAndRunning = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOn
+    Running = $script:RunningTrue
+}
+
+. $script:configurationFile -Name $Name -Server $Server -User $User -Password $Password
+
+$script:mofFileWithServicePolicyUnset = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyUnset)\"
+$script:mofFileWithServicePolicyOn = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOn)\"
+$script:mofFileWithServicePolicyOff = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOff)\"
+$script:mofFileWithServicePolicyOnAndRunning = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOnAndRunning)\"
+
+Describe "$($script:dscResourceName)_Integration" {
```

Comment:
The integration tests need little refactoring as per the documentation guidelines (Separate the Get-DscConfiguration in two separate tests).